tests/model_serving/model_server/inference_service_configuration/utils.py

```diff
@@ -0,0 +1,112 @@
+from simple_logger.logger import get_logger
+from timeout_sampler import TimeoutSampler
+
+from utilities.constants import Timeout
+from utilities.infra import get_pods_by_isvc_label
+
+LOGGER = get_logger(name=__name__)
+
+
+@contextmanager
+def update_inference_service(
+    client: DynamicClient, isvc: InferenceService, isvc_updated_dict: dict[str, Any], wait_for_new_pods: bool = True
+) -> Generator[InferenceService, Any, None]:
+    """
+    Update InferenceService object.
+
+    Args:
+        client (DynamicClient): DynamicClient object.
+        isvc (InferenceService): InferenceService object.
+        isvc_updated_dict (dict[str, Any]): InferenceService object.
+        wait_for_new_pods (bool): Whether to wait for new pods to be created.
+
+    """
+    orig_pods = get_pods_by_isvc_label(client=client, isvc=isvc)
+
+    with ResourceEditor(patches={isvc: isvc_updated_dict}):
+        if wait_for_new_pods:
+            wait_for_new_running_inference_pods(isvc=isvc, orig_pods=orig_pods)
+
+        yield isvc
+
+
+def verify_env_vars_in_isvc_pods(isvc: InferenceService, env_vars: list[dict[str, str]], vars_exist: bool) -> None:
+    """
+    Verify that the environment variables in the InferenceService pods match the expected values.
+
+    Args:
+        isvc (InferenceService): InferenceService object.
+        env_vars (list[dict[str, str]]): List of environment variables to verify.
+        vars_exist (bool): Whether the environment variables should exist in the pod.
+
+    Raises:
+        ValueError: If the environment variables do not match the expected values.
+    """
+    unset_pods = []
+    checked_env_vars_names = [env_var["name"] for env_var in env_vars]
+
+    pods = get_pods_by_isvc_label(client=isvc.client, isvc=isvc)
+
+    for pod in pods:
+        pod_env_vars_names = [env_var.name for env_var in pod.instance.spec.containers[0].get("env", [])]
+        envs_in_pod = [env_var in pod_env_vars_names for env_var in checked_env_vars_names]
+
+        if vars_exist:
+            if not all(envs_in_pod):
+                unset_pods.append(pod.name)
+
+        else:
+            if all(envs_in_pod):
+                unset_pods.append(pod.name)
+
+    if unset_pods:
+        raise ValueError(
+            f"The environment variables are {'not' if vars_exist else ''} set in the following pods: {unset_pods}"
+        )
+
+
+def wait_for_new_running_inference_pods(
+    isvc: InferenceService, orig_pods: list[Pod], expected_num_pods: int | None = None
+) -> None:
+    """
+    Wait for the inference pod to be replaced.
+
+    Args:
+        isvc (InferenceService): InferenceService object.
+        orig_pods (list): List of Pod objects.
+        expected_num_pods (int): Number of pods expected to be running. I
+            f not provided, the number of pods is expected to be len(orig_pods)
+
+    Raises:
+        TimeoutError: If the pods are not replaced.
+
+    """
+    LOGGER.info("Waiting for pods to be replaced")
```

Comment:
This function might be a good candidate for `@retry`

Reply:
for this use case i think it is clearer this way